### ISSUE TITLE
Hello, author. When I train my dataset, after a epoch, loss is Nan, LR is 000000. Why。。。。。。

### ISSUE THREAD

**Issue by zhanghongsir** [ruinmessi/ASFF#42]

**Comment by ruinmessi** [comment]
I have met the NaN loss problem when I use large lr. And this problem occurs in the first epoch only when I use FP16 training with mobilenet. 
So I suggest you disable the FP16 training if you change the backbone and set a longer warmup schedule.  BTW, make sure the right input the target in the first.

**Comment by zhanghongsir** [comment]
I didn't adjust the LR of the code. It's 0.001

**Comment by zhanghongsir** [comment]
Do I need to turn it down?

**Comment by ruinmessi** [comment]
Yes, it's worth to try first.

**Comment by zhanghongsir** [comment]
thank you very much,but I transformed my data set into VOC format, and only forecast one class. Last night, after I set LR to 0.0001, in the fourth epoch, loss is Nan. My training parameters are
checkpoint:darknet53_feature_mx.pth
Rbf:true
Asff:true
dropblock:true
use_cuda:true
The rest of the parameters are false
Do you have any suggestions?
thank you

**Comment by ruinmessi** [comment]
What's your batchsize. And you disable the sycn bn, mixup, smooth label, and other tricks? 
The loss becomes NaN in the 4th epoch, so I guess the LR is still larger. You have to carefully adjust it according to your size of datasets, your batchsize (use sycn bn if you have multiple gpus). 
If the size of your dataset is quite small, I suggest you use the pretrained ASFF weights as the initial checkpoint.

**Comment by zhanghongsir** [comment]
my batchsize=5,
My training set has 10461 pictures, val set and test set have 500 pictures, and I only use darknet53_feature_mx.pth.Those tricks have also been used. I saved the model after one epoch, and got the mean AP_50 = 0.1701
map_50:95:0.06698
Is this normal?

**Comment by ruinmessi** [comment]
You only have one gpu and the total batchsize is 5? With sycn bn, the equivalent total batch size of my model is 64 or 60 (with 4 v100 or 10 2080ti). So I think 0.0001 is still lager with batchsize=5. If you have no extra gpu, you may turn-off the random resizing and try to train it with larger batchsize.

**Comment by zhanghongsir** [comment]
I'm learning more and more during the training. I don't know why

**Comment by zhanghongsir** [comment]
Hello, if I only need to predict one class, I only modify voc_class = ('People ',) and num_class = 1 in vocdata.py. Do I need to change the network structure or other places?
Then in the process of training, the learning rate will become larger and larger. I see that the code is also calculated in this way. I don't understand why. Isn't it smaller and smaller with the increase of training times?

**Comment by zhanghongsir** [comment]
Hello, if I only need to predict one class, I only modify voc_class = ('People ',) and num_class = 1 in vocdata.py. Do I need to change the network structure or other places?
Then in the process of training, the learning rate will become larger and larger. I see that the code is also calculated in this way. I don't understand why. Isn't it smaller and smaller with the increase of training times?
I got poor results. Map50 is low

**Comment by ruinmessi** [comment]
You may take a look at the "warm-up" strategy.

**Comment by zhanghongsir** [comment]
ok ,thank you  very much

**Comment by dreamhighchina** [comment]
@zhanghongsir meet the same issue as you，do you solve it now？

**Comment by zhanghongsir** [comment]
sorry i can not get a good map.
The map becomes 0.09 in the 4th epoch

**Comment by CyanideCentral** [comment]
> @zhanghongsir meet the same issue as you，do you solve it now？

Changing BURN_IN to 0 in the .cfg file seems to solve it for me.

**Comment by pikerbright** [comment]
@zhanghongsir @ruinmessi I also met this issue when I train model to predict **only one class**. The root cause is that if we only have one class, the loss_cls need to be removed, because we only have positive samples but not any negative samples for this loss.
After I remove loss_cls, the training process can convergence normally.

**Comment by xuezhongcailian** [comment]
I also met this issue when I train model to predict ONLY ONE class, I try use overall method ,but loss still become Nan. please everyone help me, thank you

**Comment by awl-hung-pham** [comment]
> Hi @zhanghongsir @ruinmessi
> I also met this issue when I train model to predict **ONLY ONE** class. The root cause is that if we only have one class, the **loss_cls** need to be **removed**, because we only have one class samples but not any other calss samples for this loss, which will cause the loss become Nan.
> After I remove loss_cls, the training process can convergence normally.

@pikerbright Hi, removing loss_cls is that loss is changed from **"loss = loss_anchor + loss_iou + loss_l1+ loss_obj + loss_cls"** to **"loss = loss_anchor + loss_iou + loss_l1+ loss_obj"**

Thank you!

**Comment by hungthanhpham94** [comment]
> Hi @zhanghongsir @ruinmessi
> I also met this issue when I train model to predict **ONLY ONE** class. The root cause is that if we only have one class, the **loss_cls** need to be **removed**, because we only have one class samples but not any other calss samples for this loss, which will cause the loss become Nan.
> After I remove loss_cls, the training process can convergence normally.

@pikerbright Hi, removing loss_cls is that loss is changed from "loss = loss_anchor + loss_iou + loss_l1+ loss_obj + loss_cls" to "loss = loss_anchor + loss_iou + loss_l1+ loss_obj"
Thank you!

**Comment by 17868380981** [comment]
How to train single-class data, Do you need to change the filter of the link YOLO layer
